### PR TITLE
Fix integer overflow in cv::Luv2RGBinteger::process.

### PR DESCRIPTION
**Pull request by vrabaud**
For LL=49, uu=205, vv=23, we end up with x=7373056 and y=458
which overflows y*x.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work

### CODE REVIEW

File: modules/imgproc/src/color_lab.cpp [hunk 1]
```diff
@@ -3567,7 +3567,7 @@ struct Luv2RGBinteger
 
         long long int xv = ((int)up)*(long long)vp;
         int x = (int)(xv/BASE);
-        x = y*x/BASE;
+        x = ((long long int)y)*x/BASE;
```
Comment (alalek) [inline modules/imgproc/src/color_lab.cpp:3570]: Could you please add test for the case with overflow?
Existed `Imgproc_ColorLuv_Full.bitExactness` doesn't cover this code at all.

@savuor Which test should execute this "bit-exact" code (`Luv2RGBinteger`)?

Comment (vrabaud) [reply modules/imgproc/src/color_lab.cpp:3570]: I tried to create a test but it is tough: a Luv(49,205,23) gives RGB(0,103,255) without my patch and (255,0,255) with my patch. Then again, converting back to Luv we respectively get (122,80,15) and (153,157,31). We hit the white point limitation as referenced in the doc and here https://answers.opencv.org/question/205330/converting-luv-rgb-luv-results-in-different-values/

If we use https://www.easyrgb.com/en/convert.php, the results are very sensitive at this point and due to rounding, I cannot tell which value should be returned.

I guess we hit the limits of the space and there is no real answer: https://upload.wikimedia.org/wikipedia/commons/8/8d/SRGB_gamut_within_CIELCHuv_color_space_mesh.webm

Comment (vrabaud) [reply modules/imgproc/src/color_lab.cpp:3570]: Ok, I added a test to ensure it gives the same result as the float version.

Comment (savuor) [reply modules/imgproc/src/color_lab.cpp:3570]: @alalek This is `Imgproc_ColorLuv_Full.bitExactness`, already fixed by @vrabaud

Comment (savuor) [reply modules/imgproc/src/color_lab.cpp:3570]: @alalek The test you've mentioned should check this code for bit-exactness. If it's not true, then why?

Comment (vrabaud) [reply modules/imgproc/src/color_lab.cpp:3570]: @savuor : the bit exactness code has the same overflow bug https://github.com/opencv/opencv/blob/d58b5ef74b91ba68ef89f9ce8be8725d0dfcaa0d/modules/imgproc/test/test_color.cpp#L2572 (that's why I had to modify it too). The fact that I had to change the hashes shows that there is indeed a bug.

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Comment by vrabaud** [comment]
I just need to merge those two commits before it can merged right ?

**Comment by alalek** [comment]
Not needed - I will squash commits during the merge.
(currently I'm waiting for [finalization of nightly builds](http://pullrequest.opencv.org/buildbot/builders))

File: modules/imgproc/test/test_color.cpp [hunk 2]
```diff
@@ -1840,6 +1840,21 @@ TEST(Imgproc_ColorLuv, accuracy) { CV_ColorLuvTest test; test.safe_run(); }
 TEST(Imgproc_ColorRGB, accuracy) { CV_ColorRGBTest test; test.safe_run(); }
 TEST(Imgproc_ColorBayer, accuracy) { CV_ColorBayerTest test; test.safe_run(); }
 
+TEST(Imgproc_ColorLuv, Overflow_21112)
+{
+    const Size sz(107, 16);  // unaligned size to run both SIMD and generil code
```
Comment (vrabaud) [inline modules/imgproc/test/test_color.cpp:1845]: BTW, generil -> generic
